sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1010,7 +1014,16 @@ class Analyzer(
                 s"(valid range is [1, ${aggs.size}])")
           case o => o
         }
-        Aggregate(newGroups, aggs, child)
+        // If an aggregateExpression is integer, after replaced with this aggregateExpression, the
+        // groupExpression will be considered as an ordinal still, so we should filter it.
+        var filterGroups: Seq[Expression] = Nil
+        newGroups.foreach { ng =>
+          ng match {
+            case a: Alias => if (!isIntLiteral (a.child)) filterGroups :+= ng
+            case _ => filterGroups :+= ng
+          }
+        }
+        Aggregate(filterGroups, aggs, child)
```

Comment:
When the aggregates that have an empty input, if we remove all grouping expressions, this triggers the ungrouped code path (which aways returns a single row). So the query semantics are changed.

You meant this is not an issue anymore?